changelog/4288.bugfix.rst

```diff
@@ -0,0 +1,2 @@
+Fixed a bug in `~sunpy.net.dataretriever.sources.goes.XRSClient._get_url_for_timerange()` which returned incorrect URLs
+because of not using `**kwargs` in the client's `_get_overlap_urls()` method.
```

Comment:
```suggestion
Fixed a bug in `~sunpy.net.dataretriever.sources.goes.XRSClient._get_url_for_timerange` which returned incorrect URLs
because of not using ``**kwargs`` in the client's ``_get_overlap_urls()`` method.
```